### DISCUSSION TITLE
Update to Microsoft.Data.SqlClient

### DISCUSSION THREAD

**Discussion by janrybka** [rasmus/EventFlow#1022] [Ideas]
In first words (as it's my first post here) I'd like to give thanks to Rasmus and all contributors doing grate work on this project! 🙂 

As part of maintenance job in our project, we'd like to update SQL Client to Microsoft.Data namespace. To have full consistency, this namespace should be used by all components and this includes EventsFlow libraries. 

With this post, I'd like to give you an info that I'll be working on this update in EventFlow code (first change will be based on `develop-v0` as this is branch we're currently using) and hopefully soon you'll see a resulting PR.

To give some context why the change is needed:
1. System.Data.SqlClient is updated only with security update (ended on .net Core 2.1): https://www.nuget.org/packages/System.Data.SqlClient#versions-body-tab
2. All new features are added now only to Microsoft.Data.SqlClient: https://learn.microsoft.com/en-us/sql/connect/ado-net/introduction-microsoft-data-sqlclient-namespace?view=sql-server-ver16
3. Required changes are listed in porting guide: https://github.com/dotnet/SqlClient/blob/main/porting-cheat-sheet.md

If you see some problems with the idea or potential problems, please give a hint 🙂

**Comment by rasmus** [comment]
Awesome, will have a look. Also clearing up some legal stuff regarding the copyright (see the pr). Will do a release as soon as that is done

**Reply by janrybka** [reply]
I've posted first PR to `develop-v1` as this one turned out to be simplier: https://github.com/eventflow/EventFlow/pull/1024

Since my post I see that there is one wider update to `develop-v0` (https://github.com/eventflow/EventFlow/pull/1023). This one looks like it's also resolving issue with SqlClient version, but is much wider. At the point I don't know which solution is better, so I'll continue on this smaller changes scope: https://github.com/janrybka/EventFlow/pull/1/files
